docs/design/proposals/encryption-with-vault-sa.md

```diff
@@ -0,0 +1,84 @@
+# Encrypted Volumes with a ServiceAccount per Tenant
+
+## Requirements
+
+- Hashicorp Vault with Kubernetes Auth
+- each Tenant (=Namespace) has a ServiceAccount to connect to Vault
+- each Tenant can override Vault configuration options (like `BackendPath`)
+
+## Similarities with available KMS implementations
+
+Ceph-CSI provides several KMS implementations with different functionalities.
+Two implementations use Hashicorp Vault and could be used as a base for a new
+KMS implementation. Or, if changes would be minimal, a configuration option to
+one of the implementations can be added.
+
+Different KMS implementations and their configurable options can be found at
+[`csi-kms-connection-details.yaml`](https://github.com/ceph/ceph-csi/blob/devel/examples/kms/vault/csi-kms-connection-details.yaml).
```

Comment:
instead of the devel branch can we have a relative path here? if yes, we can change it applies for other places also

Reply:
oh, yes, will do!